network/wifi/wifi.c

```diff
@@ -127,6 +202,29 @@ static void wifi_init_intarface(struct wifi_desc *desc)
 		wifi_socket_recvfrom;
 }
 
+/* Callback to be submmited to the at_parser to get notification when a
+connection is created or closed */
+static void _wifi_connection_callback(void *ctx, enum at_event event,
+				      uint32_t conn_id,
+				      struct circular_buffer **cb)
+{
+	struct wifi_desc	*desc = ctx;
+	int32_t			sock_id;
+
+	sock_id = desc->conn_id_to_sock_id[conn_id];
+	if (event == AT_NEW_CONNECTION) {
```

Comment:
it would make more sense to change the state of the connection here than outside of the callback. That it is active or not since this callback does not return any error you have no method of checking if things where successful. That would simplify a lot the dispatch_event function.